### PR TITLE
Moving Icons away from CC BY-ND 3.0 licensed Icons to help with FOSS Packaging

### LINKED ISSUE #8077: Are the icons8 icons used actually CC BY-ND 3.0?
Are the icons8 icons actually CC BY-ND 3.0?

https://github.com/yuzu-emu/yuzu/blob/master/license.txt#L344-L361

This license below doesn't look CC BY-ND 3.0, did they used to be licensed that way?

https://intercom.help/icons8-7fb7577e8170/en/articles/5534926-universal-multimedia-licensing-agreement-for-icons8

---

When packaging yuzu for nixpkgs we dont distribute a cached build since it uses CC BY-ND 3.0 icons let alone whatever that custom unfree license is.

### PR DESCRIPTION
**Pull request by Docteh**
I've seen some comments stating that sharing pre-compiled binaries
is problematic for linux distributions due to the CC BY-ND 3.0 license

In this commit I am tackling the 16x16 icons

The reported status of the "yuzu Web Service" section of Configuration uses two icons, checked and failed

checked is replaced by material-design-icons check_circle
failed is replaced by material-design-icons error

Colors have been set by taking the white pngs of 18x18, cropping to 16x16, and then adjusting Curves using Paint.NET

Other tools are available.

There is also a locked icon that does not appear to be used in the current code. citra uses it for multiplayer

lock is replaced by material-designs-icons lock

Some of the colorful theme icons are clearly from https://github.com/icons8/flat-color-icons/ which is available MIT licensed

Sources for MIT license:
https://github.com/icons8/flat-color-icons/issues/5
https://github.com/icons8/flat-color-icons/blob/master/package.json

This covers plus_folder, bad_folder, chip, folder

Leaving plus, sd_card and star to need further research/replacement

The first two are user visible, the lock icons aren't used right now. 
image

The lock icons are shoddily replaced to avoid having to test and verify that removal doesn't break anything. 

Fixes #8077

### CODE REVIEW

File: dist/license.md [hunk 1]
```diff
@@ -2,9 +2,9 @@ The icons in this folder and its subfolders have the following licenses:
 
 Icon Name | License | Origin/Author
 --- | --- | ---
-qt_themes/default/icons/16x16/checked.png | CC BY-ND 3.0 | https://icons8.com
-qt_themes/default/icons/16x16/failed.png | CC BY-ND 3.0 | https://icons8.com
-qt_themes/default/icons/16x16/lock.png | CC BY-ND 3.0 | https://icons8.com
+qt_themes/default/icons/16x16/checked.png |Apache 2.0 | https://github.com/google/material-design-icons/
+qt_themes/default/icons/16x16/failed.png | Apache 2.0 | https://github.com/google/material-design-icons/
+qt_themes/default/icons/16x16/lock.png | Apache 2.0 | https://github.com/google/material-design-icons/
```
Comment (Atemu) [inline dist/license.md:7]: Might not be a bad idea to mention that these were modified

Comment (Docteh) [reply dist/license.md:7]: Just slap `(modified)` after the url? or more details?

**Comment by Tachi107** [comment]
I find the checked.png icon hard to look at with a white background. Would it be possible to darken its green a bit?

File: dist/license.md [hunk 2]
```diff
@@ -22,14 +22,14 @@ qt_themes/qdarkstyle/icons/48x48/folder.png | CC BY-ND 3.0 | https://icons8.com
...
-qt_themes/colorful/icons/16x16/lock.png | CC BY-ND 3.0 | https://icons8.com
+qt_themes/colorful/icons/16x16/lock.png | Apache 2.0 | https://github.com/google/material-design-icons/
 qt_themes/colorful/icons/16x16/view-refresh.png | Apache 2.0 | https://material.io
-qt_themes/colorful/icons/256x256/plus_folder.png | CC BY-ND 3.0 | https://icons8.com
-qt_themes/colorful/icons/48x48/bad_folder.png | CC BY-ND 3.0 | https://icons8.com
-qt_themes/colorful/icons/48x48/chip.png | CC BY-ND 3.0 | https://icons8.com
-qt_themes/colorful/icons/48x48/folder.png | CC BY-ND 3.0 | https://icons8.com
+qt_themes/colorful/icons/256x256/plus_folder.png | MIT | https://github.com/icons8/flat-color-icons
+qt_themes/colorful/icons/48x48/bad_folder.png | MIT | https://github.com/icons8/flat-color-icons
+qt_themes/colorful/icons/48x48/chip.png | MIT | https://github.com/icons8/flat-color-icons
```
Comment (Tachi107) [inline dist/license.md:29]: The `chip.png` shipped by yuzu is similar to the [`electronics.svg`](https://github.com/icons8/flat-color-icons/blob/8eccbbbd8b2af1d2c9593e7cfba5ecb0d68ee378/svg/electronics.svg) you can find in the flat-color-icons pack, but its not the same image

File: dist/license.md [hunk 3]
```diff
@@ -22,14 +22,14 @@ qt_themes/qdarkstyle/icons/48x48/folder.png | CC BY-ND 3.0 | https://icons8.com
...
 qt_themes/qdarkstyle/icons/48x48/star.png | CC BY-ND 3.0 | https://icons8.com
-qt_themes/colorful/icons/16x16/lock.png | CC BY-ND 3.0 | https://icons8.com
+qt_themes/colorful/icons/16x16/lock.png | Apache 2.0 | https://github.com/google/material-design-icons/
 qt_themes/colorful/icons/16x16/view-refresh.png | Apache 2.0 | https://material.io
-qt_themes/colorful/icons/256x256/plus_folder.png | CC BY-ND 3.0 | https://icons8.com
-qt_themes/colorful/icons/48x48/bad_folder.png | CC BY-ND 3.0 | https://icons8.com
-qt_themes/colorful/icons/48x48/chip.png | CC BY-ND 3.0 | https://icons8.com
-qt_themes/colorful/icons/48x48/folder.png | CC BY-ND 3.0 | https://icons8.com
+qt_themes/colorful/icons/256x256/plus_folder.png | MIT | https://github.com/icons8/flat-color-icons
+qt_themes/colorful/icons/48x48/bad_folder.png | MIT | https://github.com/icons8/flat-color-icons
```
Comment (Tachi107) [inline dist/license.md:28]: `bad_folder.png` and `plus_folder.png` are not from flat-color-icons, but are similar to [`folder.svg`](https://github.com/icons8/flat-color-icons/blob/8eccbbbd8b2af1d2c9593e7cfba5ecb0d68ee378/svg/folder.svg)

File: dist/license.md [hunk 4]
```diff
@@ -22,14 +22,14 @@ qt_themes/qdarkstyle/icons/48x48/folder.png | CC BY-ND 3.0 | https://icons8.com
 qt_themes/qdarkstyle/icons/48x48/plus.png | CC0 1.0 | Designed by BreadFish64 from the Citra team
 qt_themes/qdarkstyle/icons/48x48/sd_card.png | CC BY-ND 3.0 | https://icons8.com
 qt_themes/qdarkstyle/icons/48x48/star.png | CC BY-ND 3.0 | https://icons8.com
-qt_themes/colorful/icons/16x16/lock.png | CC BY-ND 3.0 | https://icons8.com
+qt_themes/colorful/icons/16x16/lock.png | Apache 2.0 | https://github.com/google/material-design-icons/
```
Comment (Tachi107) [inline dist/license.md:25]: `lock.png` is already MIT licensed, see [`lock.svg`](https://github.com/icons8/flat-color-icons/blob/8eccbbbd8b2af1d2c9593e7cfba5ecb0d68ee378/svg/lock.svg)

**Review by Tachi107 — CHANGES_REQUESTED** [review]

File: dist/license.md (hunk 3 above)
Comment (Docteh) [reply dist/license.md:28]: Yeah, modifications are open source, but its not clear if it'd stay MIT, or become GPLv2. 

Found a comment that mentions the modifier of the icons has signed yuzu CLA https://github.com/yuzu-emu/yuzu/pull/2898#issuecomment-538192851

**Comment by Docteh** [comment]
I'm not actually resizing the images, its a crop
image


Retint of the checked.png, I grabbed the color from the current checked.png
image

I'm going to convert this to a draft, just to avoid builds being generated when I push

File: dist/license.md (hunk 3 above)
Comment (Atemu) [reply dist/license.md]: I don't think including icon graphics in a GPL'd software triggers a GPL infection. It'd stay MIT unless the right holder of the modification decided to re-license under an MIT-compatible license such as the GPLv2.

IANAL.

**Comment by Docteh** [comment]
Current icons
image


https://remixicon.com/ Apache 2.0 
image

Hero Icons, MIT https://github.com/tailwindlabs/heroicons

This one looks closer to what icons8 has. It doesn't have an SD Card, but that can be gotten to.
image

bad_folder

Personally, I lean towards liking heroicons

@Tachi107 they have a LOT of icons to look through. 

- [ ] are heroicons good enough?
- [ ] sd cards

**Comment by Docteh** [comment]
I've got all the ND icons replaced, so I squashed it down, and have double checked the changes to `dist/license.md`

Previous Colorful
image

sd_card, star, plus changed in this PR
image


New White icons
image

**Comment by Docteh** [comment]
Two changes, that I'll commit later. I don't want to spam builds. (These are the 48x48 pngs that will be commited later)

Star, Original color: EFCE4A different color: FFCA28 Set size to 42
star


SD Card is also a bit big, initially rasterized 48x48, better off at 44x44
sd_card

Current PR/Preview with these changes
imageimage

**Comment by Docteh** [comment]
For the monochromatic icons I'm not sure if they should be filled in or not, but after staring at a lot of icons I believe they should be the same

Current
image

This PR
image

**Review by Tachi107 — APPROVED** [review]

**Review by IvarWithoutBones — APPROVED** [review]
We have been using a [rehosted patch](https://github.com/NixOS/nixpkgs/blob/e4d49de45a3b5dbcb881656b4e3986e666141ea9/pkgs/applications/emulators/yuzu/yuzu-free-icons.patch) of this PR for a few months now in [nixpkgs](https://github.com/NixOS/nixpkgs), as it allows us to redistribute binaries instead of having the end user compile it themselves. Redistributing the official binaries is not an option for us on NixOS, as our dynamic linker doesn't play nicely with that.

 It would be great if this could get merged!

**Comment by Docteh** [comment]
#8541 adds 11 more icons, in that, 3 variants of 3, and 2 variants of 1 add up to 11. I complained over there about the issue, might need a reminder.

I got feedback from @german77 that the SD card looks like a foot, changed it out. Still a few pins short, but at 48x48 accuracy might he bard to see. 
image

Redrew plus signs, the green felt too small, the others felt too big. I had a hard time seeing the exclamation mark in `failed`, so it's also replaced
plus plus plus failed

**Comment by Docteh** [comment]
@IvarWithoutBones Looks like #8541 got merged  https://github.com/yuzu-emu/yuzu/commit/705f7db84dd85555a6aef1e136cf251725cef293 so I'll need to dedicate some time for those.

**Comment by IvarWithoutBones** [comment]
Thanks for the hard work and keeping us updated, its much appreciated ☺️

**Comment by Docteh** [comment]
Big August Update!

... I reworked the changes to `dist/license.md` (now retired) to the new `.reuse/dep5` file


#8541 also adds some other icons, so

Remaining NO DERIVATIVES 
```
Files: dist/qt_themes/*/icons/16x16/connected.png
       dist/qt_themes/*/icons/16x16/connected_notification.png
       dist/qt_themes/*/icons/16x16/disconnected.png
       dist/qt_themes/*/icons/48x48/no_avatar.png
Copyright: https://icons8.com
License: CC-BY-ND-3.0
```

no_avatar candidates, https://ionic.io/ionicons https://github.com/ionic-team/ionicons (person)
no_avatar no_avatar

**Comment by Docteh** [comment]
@Tachi107 I copy/pasted the Creative Commons Attribution-ShareAlike 3.0 Unported license into notepad. Seems okay. But maybe there is better source with files in .txt format?

**Comment by Tachi107** [comment]
> @Tachi107 I copy/pasted the Creative Commons Attribution-ShareAlike 3.0 Unported license into notepad. Seems okay. But maybe there is better source with files in .txt format?

Yes, there is. You can use `reuse download --all` to download all the licenses used by the project into the  'LICENSES/` folder.

You can install the `reuse` tool on Debian with `apt install reuse`, or via pip with `pip3 install reuse`

**Comment by Docteh** [comment]
I swapped out the license with the .txt direct from Creative Commons. Why? easier to read on GitHub, or in less. 

Goes well with CC0-1.0.txt, also nicely formatted.

**Comment by Tachi107** [comment]
> I swapped out the license with the .txt direct from Creative Commons. Why? easier to read on GitHub, or in less.

Yeah I guess that is fine too. If the reuse CI job passes it means that everything is OK

**Review by liamwhite — APPROVED** [review]

**Comment by Tachi107** [comment]
Thanks for merging!

**Pull request merged**